src/code_182B30.c

```diff
@@ -175,7 +175,22 @@ INCLUDE_ASM(s32, "code_182B30", func_802597B0);
 
 INCLUDE_ASM(s32, "code_182B30", func_8025995C);
 
-INCLUDE_ASM(s32, "code_182B30", func_80259A48);
+void func_80259A48(s32 arg0, ActorPart* arg1, s32 arg2, s32 arg3) {
+    DecorationTable* decorationTable = arg1;
```

Comment:
is this assignment required?

Reply:
I was having issues without it

Reply:
You're assigning an ActorPart* to a DecorationTable* though. if you can't find a way to clean this up, could you leave a comment saying "TODO look into" or something?